wandb/sdk/launch/agent/agent.py

```diff
@@ -570,21 +599,7 @@ def _thread_run_job(
         self._set_queue_and_rqi_in_project(project, job, job_tracker.queue)
         api.ack_run_queue_item(job["runQueueItemId"], project.run_id)
         # don't launch sweep runs if the sweep isn't healthy
-        if launch_spec.get("sweep_id"):
-            try:
-                state = api.get_sweep_state(
-                    sweep=launch_spec["sweep_id"],
-                    entity=launch_spec["entity"],
-                    project=launch_spec["project"],
-                )
-            except Exception as e:
-                _logger.debug(f"Fetch sweep state error: {e}")
-                state = None
-
-            if state != "RUNNING" and state != "PAUSED":
-                raise LaunchError(
-                    f"Launch agent picked up sweep job, but sweep ({launch_spec['sweep_id']}) was in a terminal state ({state})"
-                )
+        self.check_sweep_state(launch_spec, api)
```

Comment:
moved this into the new `check_sweep_state` helper because this function was a little too complex for ruff's basic ass